docs/dependency.rst

```diff
@@ -0,0 +1,245 @@
+  |   |   `-- contrast.py
+  |   `-- __init__.py
+  |-- README.md
+  `-- setup.py
+
+Using pysat to test your instruments
+------------------------------------
+
+A generalized instrument test class is provided under ``pysat.tests`` for
+developers.  Continuing the above example, developers may copy over the
+``test_instruments.py`` file and update it in a few locations.  For example
+
+.. code:: python
+
+  # Make sure to import your instrument library here
+  import customLibrary
+
+  # Import the test classes from pysat
+  from pysat.tests.instrument_test_class import generate_instrument_list
+  from pysat.tests.instrument_test_class import InstTestClass
+
+  # Developers for instrument libraries should update the following line to
+  # point to their own library location. For example,
+  # instruments = generate_instrument_list(inst_loc=mypackage.instruments)
+  instruments = generate_instrument_list(inst_loc=customLibrary.instruments)
+
+The above code scans the list of instruments and flags each instrument for one
+or more of the test types, as defined below.  This bit of the code should
+generally be unchanged.  Instruments are grouped in three lists:
+
+* instruments['names']: A list of all module names to check for
+  standardization
+* instruments['download']: A list of dicts containing info to initialize
+  instruments for end-to-end testing
+* instruments['no_download']: A list of dicts containing info to initialize
+  instruments without download support for specialized local tests
+
+.. code:: python
+
+  # The following lines apply the custom instrument lists to each type of test
+  method_list = [func for func in dir(InstTestClass)
+                 if callable(getattr(InstTestClass, func))]
+  # Search tests for iteration via pytestmark, update instrument list
+  for method in method_list:
+      if hasattr(getattr(InstTestClass, method), 'pytestmark'):
+          # Get list of names of pytestmarks
+          nargs = len(getattr(InstTestClass, method).pytestmark)
+          names = [getattr(InstTestClass, method).pytestmark[j].name
+                   for j in range(0, nargs)]
+          # Add instruments from your library
+          if 'all_inst' in names:
+              mark = pytest.mark.parametrize("inst_name", instruments['names'])
+              getattr(InstTestClass, method).pytestmark.append(mark)
+          elif 'download' in names:
+              mark = pytest.mark.parametrize("inst_dict",
+	                                        instruments['download'])
+              getattr(InstTestClass, method).pytestmark.append(mark)
+          elif 'no_download' in names:
+              mark = pytest.mark.parametrize("inst_dict",
+                                             instruments['no_download'])
+              getattr(InstTestClass, method).pytestmark.append(mark)
+
+Finally, the ``setup`` function under the ``TestInstruments`` class should be
+updated with the location of the instrument subpackage.
```

Comment:
```suggestion
Finally, the ``setup_class`` function under the ``TestInstruments`` class should be
updated with the location of the instrument subpackage.  Note that the routine uses 
temporary directories to store downloaded files to avoid breaking user's directory 
structure.
```